user_data.sh

```diff
@@ -27,31 +27,101 @@ fi
 # we know about from introspection.
 #
 network_config_files() {
-    if [ "$OS_TYPE" = "centos" ] ; then
+if [ "$OS_TYPE" = "centos" ] ; then
 cat << EOF
-write_files:
-- path: /etc/sysconfig/network-scripts/ifcfg-eth1
-  owner: root:root
-  permissions: '0644'
-  content: |
-    BOOTPROTO=dhcp
-    DEVICE=eth1
-    ONBOOT=yes
-    TYPE=Ethernet
-    USERCTL=no
+hostname: dev
+users:
+  - name: metal3
+    groups: wheel
+    sudo: ['ALL=(ALL) NOPASSWD:ALL']
+    lock_passwd: false
+    shell: /bin/bash
+    ssh-authorized-keys:
+      - <key>
+
+yum_repos:
+    kubernetes:
+        baseurl: https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64
+        enabled: 1
+        gpgcheck: 1
+        repo_gpgcheck: 1
+        gpgkey: https://packages.cloud.google.com/yum/doc/yum-key.gpg https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+
 runcmd:
- - [ ifup, eth1 ]
+  - [ ifup, eth1 ]
+  # Install updates
+  - yum check-update
+  # Install keepalived
+  - yum install -y gcc kernel-headers kernel-devel
+  - yum install -y keepalived
+  - systemctl start keepalived
+  - systemctl enable keepalived
+  # Install docker
+  - yum install -y yum-utils device-mapper-persistent-data lvm2
+  - yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+  - yum install docker-ce docker-ce-cli containerd.io -y
+  - usermod -aG docker metal3
+  - systemctl start docker
+  - systemctl enable docker
+  # Install kubernetes
+  - setenforce 0
+  - sed -i 's/^SELINUX=enforcing$/SELINUX=permissive/' /etc/selinux/config
+  - yum install -y kubelet kubeadm kubectl --disableexcludes=kubernetes
+  - systemctl enable --now kubelet
+  - kubeadm init --apiserver-advertise-address 55.66.77.88 --ignore-preflight-errors=all 
```

Comment:
some kubeadm init options:
--upload-certs could be included here
--certificate-key to pass a self generated key

kubeadm join to be included here with a self-generated token. I can provide the details.